File: Python/Method.py
import os               # Operating System specific functionality
import sys              # System-specific parameters and functions
import math             # Mathematical functions
import random           # Generate pseudo-random numbers
import datetime         # Basic date and time types
import string           # Common string operations
import json             # JSON encoder and decoder
import csv              # CSV file reading and writing
import urllib.request   # URL handling module
import re               # Regular expression operations
import subprocess       # Subprocess management
import shutil           # High-level file operations
import itertools        # Functions creating iterators for efficient looping

# Method 1: Print to console
def print_example():
    print("Hello, World!")

# Method 2: Read from console
def input_example():
    name = input("Enter your name: ")
    print("Hello,", name)

# Method 3: Check if file exists
def file_exists_example():
    if os.path.exists("example.txt"):
        print("File exists!")
    else:
        print("File does not exist!")

# Method 4: Write to file
def write_file_example():
    with open("output.txt", "w") as file:
        file.write("This is a test.")

# Method 5: Read from file
def read_file_example():
    with open("input.txt", "r") as file:
        content = file.read()
        print("Content of input file:", content)

# Method 6: Math operations
def math_example():
    num = math.sqrt(16)
    print("Square root of 16:", num)

# Method 7: Generate random number
def random_example():
    num = random.randint(1, 100)
    print("Random number between 1 and 100:", num)

# Method 8: Get current date and time
def datetime_example():
    now = datetime.datetime.now()
    print("Current date and time:", now)

# Method 9: String operations
def string_example():
    s = "Hello, World!"
    print("Length of string:", len(s))
    print("Uppercase:", s.upper())
    print("Lowercase:", s.lower())
    print("Capitalized:", s.capitalize())
    print("Reversed:", s[::-1])

# Method 10: JSON encoding and decoding
def json_example():
    data = {'name': 'John', 'age': 30}
    json_str = json.dumps(data)
    print("JSON string:", json_str)

# Method 11: CSV file operations
def csv_example():
    with open('data.csv', mode='w', newline='') as file:
        writer = csv.writer(file)
        writer.writerow(['Name', 'Age'])
        writer.writerow(['John', 30])

# Method 12: URL handling
def urllib_example():
    with urllib.request.urlopen("https://www.example.com") as response:
        html = response.read()
        print("HTML content:", html)

# Method 13: Regular expression operations
def re_example():
    text = "The quick brown fox jumps over the lazy dog."
    pattern = r"\b\w{3}\b"
    matches = re.findall(pattern, text)
    print("Matches:", matches)

# Method 14: Subprocess management
def subprocess_example():
    result = subprocess.run(["ls", "-l"], capture_output=True, text=True)
    print("List of files:", result.stdout)

# Method 15: Copy files
def shutil_example():
    shutil.copy("source.txt", "destination.txt")
    print("File copied successfully.")

# Method 16: Iterate through combinations
def itertools_example():
    letters = ['a', 'b', 'c']
    combos = itertools.combinations(letters, 2)
    print("Combinations:", list(combos))

# Method 17: Iterate through permutations
# (Permutations example already used in itertools_example)

# Method 18: Iterate through product
def itertools_product_example():
    numbers = [1, 2]
    letters = ['a', 'b']
    products = itertools.product(numbers, letters)
    print("Product:", list(products))

# Method 19: Iterate through cycle
def itertools_cycle_example():
    colors = ['red', 'green', 'blue']
    cycle = itertools.cycle(colors)
    print("Cycling through colors:")
    for _ in range(5):
        print(next(cycle))

# Method 20: Iterate through count
def itertools_count_example():
    count = itertools.count(start=10, step=2)
    print("Counting:")
    for _ in range(5):
        print(next(count))

# Call each method to demonstrate
print_example()
input_example()
file_exists_example()
write_file_example()
read_file_example()
math_example()
random_example()
datetime_example()
string_example()
json_example()
csv_example()
urllib_example()
re_example()
subprocess_example()
shutil_example()
itertools_example()
itertools_product_example()
itertools_cycle_example()
itertools_count_example()

File: Javascript/Method.js
// Method 1: Output to console
function consoleLogExample() {
    console.log("Hello, World!");
}

// Method 2: Read from console
function promptExample() {
    var name = prompt("Enter your name:");
    console.log("Hello, " + name);
}

// Method 3: Alert dialog
function alertExample() {
    alert("This is an alert!");
}

// Method 4: Check if element exists
function getElementByIdExample() {
    var element = document.getElementById("exampleElement");
    if (element) {
        console.log("Element exists!");
    } else {
        console.log("Element does not exist!");
    }
}

// Method 5: Set innerHTML of element
function setInnerHTMLExample() {
    var element = document.getElementById("exampleElement");
    if (element) {
        element.innerHTML = "This is new content.";
    } else {
        console.log("Element does not exist!");
    }
}

// Method 6: Get element by class name
function getElementsByClassNameExample() {
    var elements = document.getElementsByClassName("exampleClass");
    console.log("Number of elements with class 'exampleClass':", elements.length);
}

// Method 7: Create element
function createElementExample() {
    var newElement = document.createElement("div");
    newElement.innerHTML = "This is a new element.";
    document.body.appendChild(newElement);
}

// Method 8: Math operations
function mathExample() {
    var num = Math.sqrt(16);
    console.log("Square root of 16:", num);
}

// Method 9: Generate random number
function randomExample() {
    var num = Math.floor(Math.random() * 100) + 1;
    console.log("Random number between 1 and 100:", num);
}

// Method 10: Get current date and time
function dateTimeExample() {
    var now = new Date();
    console.log("Current date and time:", now);
}

// Method 11: String operations
function stringExample() {
    var str = "Hello, World!";
    console.log("Length of string:", str.length);
    console.log("Uppercase:", str.toUpperCase());
    console.log("Lowercase:", str.toLowerCase());
    console.log("Substring:", str.substring(0, 5));
}

// Method 12: JSON parsing
function jsonExample() {
    var jsonString = '{"name": "John", "age": 30}';
    var jsonObject = JSON.parse(jsonString);
    console.log("Parsed JSON object:", jsonObject);
}

// Method 13: AJAX request
function ajaxExample() {
    var xhttp = new XMLHttpRequest();
    xhttp.onreadystatechange = function() {
        if (this.readyState == 4 && this.status == 200) {
            console.log("Response:", this.responseText);
        }
    };
    xhttp.open("GET", "example.txt", true);
    xhttp.send();
}

// Method 14: Regular expression operations
function regexExample() {
    var str = "The quick brown fox jumps over the lazy dog.";
    var pattern = /\b\w{3}\b/g;
    var matches = str.match(pattern);
    console.log("Matches:", matches);
}

// Method 15: Local storage
function localStorageExample() {
    localStorage.setItem("username", "John");
    var username = localStorage.getItem("username");
    console.log("Username from local storage:", username);
}

// Method 16: Session storage
function sessionStorageExample() {
    sessionStorage.setItem("token", "abcd1234");
    var token = sessionStorage.getItem("token");
    console.log("Token from session storage:", token);
}

// Method 17: Set timeout
function setTimeoutExample() {
    setTimeout(function() {
        console.log("Timeout completed.");
    }, 2000);
}

// Method 18: Set interval
function setIntervalExample() {
    var counter = 0;
    var interval = setInterval(function() {
        console.log("Interval:", counter++);
        if (counter > 5) {
            clearInterval(interval);
        }
    }, 1000);
}

// Method 19: Fetch API
function fetchExample() {
    fetch('example.txt')
        .then(response => response.text())
        .then(data => console.log("Fetch response:", data))
        .catch(error => console.error("Fetch error:", error));
}

// Method 20: Event handling
function eventListenerExample() {
    document.getElementById("exampleButton").addEventListener("click", function() {
        console.log("Button clicked!");
    });
}

// Call each method to demonstrate
consoleLogExample();
promptExample();
alertExample();
getElementByIdExample();
setInnerHTMLExample();
getElementsByClassNameExample();
createElementExample();
mathExample();
randomExample();
dateTimeExample();
stringExample();
jsonExample();
ajaxExample();
regexExample();
localStorageExample();
sessionStorageExample();
setTimeoutExample();
setIntervalExample();
fetchExample();
eventListenerExample();

